pkg/validator/validator.go

```diff
@@ -12,15 +12,187 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package validator provides functions to validate jira issue against
+// validation criteria.
 package validator
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
 )
 
-type Validator struct{}
+const (
+	// jiraResponseSizeLimitBytes is the maximum bytes be read from JIRA REST
+	// API response.
+	jiraResponseSizeLimitBytes = 4_000_000 // 4mb
+)
+
+// Validator validates jira issue against validation criteria.
+type Validator struct {
+	// baseURL is the JIRA REST API url. Example:
+	//     https://your-domain.atlassian.net/rest/api/3
+	baseURL *url.URL
+
+	// httpClient is an HTTP client used for making outbound requests.
+	httpClient *http.Client
+
+	// jiraAccount is the user name used in [JIRA Basic Auth].
+	// [JIRA Basic Auth]: https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/
+	jiraAccount string
```

Comment:
nit: might as well call it `account` since you don't prefix other fields with `jira`